src/kirby/components/modal/services/modal.controller.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable, ViewContainerRef } from '@angular/core';
+
+import { ModalHelper } from './modal.helper';
+import { ModalConfig } from '../config/modal-config';
+import { IModalController } from './modal.controller.interface';
+
+@Injectable()
+export class ModalController implements IModalController {
+  private modals: { close: (data?: any) => {} }[] = [];
+
+  constructor(private modalHelper: ModalHelper) {}
+
+  public showModal(
+    config: ModalConfig,
+    vcRef: ViewContainerRef,
+    onCloseModal?: (data?: any) => any
```

Comment:
Maybe just `void` as return type?
```suggestion
    onCloseModal?: (data?: any) => void
```